randovania/games/am2r/generator/bootstrap.py

```diff
@@ -76,16 +76,37 @@ def assign_pool_results(self, rng: Random, patches: GamePatches, pool_results: P
         locations = all_dna_locations(patches.game, config)
         rng.shuffle(locations)
 
-        all_dna = [pickup for pickup in list(pool_results.to_place)
-                   if pickup.pickup_category is METROID_DNA_CATEGORY]
-        if len(all_dna) > len(locations):
+        total_dna = [pickup for pickup in list(pool_results.to_place + pool_results.starting)
+                     if pickup.pickup_category is METROID_DNA_CATEGORY]
+
+        dna_to_place = [pickup for pickup in list(pool_results.to_place)
```

Comment:
This list is only ever used for the length. This could be a `sum(1 for ...)` instead. Would also use a different variable name.
Also the `list` is redundant.